core/src/services/swift/core.rs

```diff
@@ -29,6 +29,7 @@ pub struct SwiftCore {
     pub root: String,
     pub endpoint: String,
     pub account: String,
+    pub storage_url: String,
```

Comment:
It's a bit confused that user need to set two endpoint for storage. How about fecth them via `auth`?

Reply:
The user needs to choose one of these methods for configuration. I attempt to explain it in the documentation, but it does add a bit of confusion for the user.

If we introduce the authentication method in the code, we need the user to add more configuration parameters. These are the three parameters needed for the authentication process: `X-Auth-User` (different from `account`), `X-Auth-Key`, `auth_url`.

```shell
curl -v -i \
  -H 'X-Auth-User: test:tester' \
  -H 'X-Auth-Key: testing' \
  http://127.0.0.1:8080/auth/v1.0
```

If we use these three parameters, we can avoid users to configure `endpoint`, `account`, `storage_url`. But I'm not quite sure whether `X-Auth-User` and `X-Auth-Key`  are intuitive to the user.

Reply:
Ok, I got this. Sorry for mixing the auth function here.

---

Since `account` is only used for building request url. I think we can change in this way.

- Change the definition of `endpoint`: it should be a valid storage URL. We can provide some examples for it.
- Remove the need of `account`: We only need `token` for auth.

So users should figure the correct endpoint and token to use for swift service.

Reply:
Thanks a lot for your suggestion! This will appear very clear. I will make the modifications later.